pkg/info/infopb/rpc.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+syntax = "proto3";
+package thanos;
+
+import "store/labelpb/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "infopb";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint and opening a door
+// for zero-copy casts to/from prometheus data types.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+// Info represents the API that is responsible for gathering metadata about the all APIs supported by the component.
+service Info {
+    // Info returns the metadata (Eg. LabelSets, Min/Max time) about all the APIs the component supports.
+    rpc Info(InfoRequest) returns (InfoResponse);
+}
+
+message InfoRequest {}
+
+enum ComponentType {
+    UNKNOWN = 0;
+    QUERY = 1;
+    RULE = 2;
+    SIDECAR = 3;
+    STORE = 4;
+    RECEIVE = 5;
+    // DEBUG represents some debug StoreAPI components e.g. thanos tools store-api-serve.
+    DEBUG = 6;
+  }
+
+message InfoResponse {
+    repeated ZLabelSet label_sets = 1 [(gogoproto.nullable) = false];
+    ComponentType ComponentType  = 2;
+    
+    StoreInfo storeInfo = 3;
+    RulesInfo rulesInfo = 4;
+    MetadataInfo metadataInfo = 5;
+    ExemplarsInfo exemplarsInfo = 6;
+    TargetsInfo targetsInfo = 7;
+}
+
+message StoreInfo {
+    int64 min_time = 1;
+    int64 max_time = 2;
+}
+
+message RulesInfo {
+    int64 min_time = 1;
+    int64 max_time = 2;
```

Comment:
What does min max time mean for rules, metadata and targets info?

Reply:
Yes. It doesn't make any sense 🤦🏻‍♂️. We always use current rules.